server/src/index.ts

```diff
@@ -298,13 +315,66 @@ app.get("/api/searchPublicActivities", async (req: Request, res: Response) => {
   });
 });
 
-app.get("/api/loadPromotedContent", (req: Request, res: Response) => {
-  res.send({
-    success: true,
-    carouselData: {},
-  });
+app.get("/api/loadPromotedContent", async (req: Request, res: Response) => {
+  const loggedInUserId = Number(req.cookies.userId);
+  const isAdmin = loggedInUserId ? await getIsAdmin(loggedInUserId) : false;
```

Comment:
Move this check for `isAdmin` into `loadPromotedContent` and pass in `loggedInUserId`. (Same for other functions.)